docs/flattening_table.md

```diff
@@ -0,0 +1,208 @@
+do not include the additional `t.call AS call` statement in the `FROM` clause.
+Please see
+[this page](https://cloud.google.com/bigquery/docs/reference/standard-sql/migrating-from-legacy-sql#removing_repetition_with_flatten)
+for more details. Also, note that explicitly using `UNNEST` is not necessary as
+a fully-qualified path is used, but you may also use `UNNEST(call) AS call`
+instead of `t.call AS call`. Please see
+[here](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#field_path)
+for more details.
+
+You can include additional information for each call by adding them to the
+`SELECT` clause. For instance, the following query adds the call genotypes (as
+an array of integers) to the result.
+
+```
+#standardSQL
+SELECT
+  reference_name, start_position, end_position, reference_bases,
+  call.name AS call_name, call.genotype
+FROM
+  `project.dataset.table` AS t,
+  t.call AS call
+```
+
+![Flatten call orig genotype](images/flatten_call_orig_genotype.png)
+
+To further flatten the BigQuery table on the genotype array (i.e. have one
+genotype per row), you can add another explicit join with `call.genotype` as
+follows:
+
+```
+#standardSQL
+SELECT
+  reference_name, start_position, end_position, reference_bases,
+  call.name AS call_name, genotype
+FROM
+  `project.dataset.table` AS t,
+  t.call AS call,
+  call.genotype as genotype
+```
+
+![Flatten call, genotype](images/flatten_call_flatten_genotype.png)
+
+Note that in this case, the call names are duplicated as each call contains
+two genotype values.
+
+Let's add `alternate_bases` to the `SELECT` clause, which is an independently
+repeated record:
+
+```
+#standardSQL
+SELECT
+  reference_name, start_position, end_position, reference_bases,
+  call.name AS call_name, genotype, alternate_bases
+FROM
+  `project.dataset.table` AS t,
+  t.call AS call,
+  call.genotype as genotype
+```
+
+![Flatten call, genotype orig alt](images/flatten_call_flatten_genotype_orig_alt.png)
+
+This result looks odd as it contains both `alternate_bases` even though
+we have flattened the `genotype` column. We really only want to return the
+particular alternate base that matches the index of the `genotype` column. This
+can be done using `ORDINAL` as follows:
+
+```
+#standardSQL
+SELECT
+  reference_name, start_position, end_position, reference_bases,
+  call.name AS call_name, genotype,
+  IF(genotype > 0, alternate_bases[ORDINAL(genotype)], NULL) AS alternate_bases
+FROM
+  `project.dataset.table` AS t,
+  t.call AS call,
+  call.genotype as genotype
+```
+
+![Flatten call, genotype, alt](images/flatten_call_flatten_genotype_flatten_alt.png)
+
+Note that the semantics of the value of the `genotype` column is has changed
+as each row only contains a single alternate allele. As a result, you may
+decide to reformat that column using
+`IF(genotype > 0, 1, genotype) AS alt_genotype`, which results to:
+  * `0` implying reference match.
+  * `-1` implying not called.
```

Comment:
Where does `-1` come from?

Reply:
It actually comes from the GA4GH API. I realize that we should update those as part of the documentation update work in Q3 as well (likely in the "understanding the BQ schema" doc). For now, I just added a note about it.

Reply:
I see, thanks for clarification.